src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java

```diff
@@ -137,7 +141,13 @@ private void updateFlowPathsStateForFlow(SwitchId switchId, int port, List<FlowP
                         || (pathSegment.getDestPort() == port
                         && switchId.equals(pathSegment.getDestSwitch().getSwitchId()))) {
                     pathSegment.setFailed(true);
-                    pathSegmentRepository.updateFailedStatus(fp.getPathId(), pathSegment, true);
+                    try {
+                        pathSegmentRepository.updateFailedStatus(fp.getPathId(), pathSegment, true);
+                    } catch (EntityNotFoundException e) {
+                        log.warn("FlowPath {} not found. Skipping reroute for flow {}.",
+                                fp.getPathId(), fp.getFlow().getFlowId(), e);
+                        return false;
```

Comment:
em... what about the rest of paths passed into `paths` argument? Now you stop processing paths on first error (not so critical as stop processing all flows on the first error, but still incorrect).

Reply:
Reworked.